include/MixerProfiler.h

```diff
@@ -41,21 +42,34 @@ class MixerProfiler
 		m_periodTimer.reset();
 	}
 
-	void finishPeriod( sample_rate_t sampleRate, fpp_t framesPerPeriod );
+	void finishPeriod(sample_rate_t sampleRate, fpp_t framesPerPeriod);
 
 	int cpuLoad() const
 	{
 		return m_cpuLoad;
 	}
 
-	void setOutputFile( const QString& outputFile );
+	void setOutputFile(const QString& outputFile);
 
+	void startDetail(const unsigned int index) { m_detailTimer[index].reset(); }
+	void finishDetail(const unsigned int index) { m_detailTime[index] = m_detailTimer[index].elapsed(); }
+
+	int detailLoad(const unsigned int index) const { return m_detailLoad[index]; }
 
 private:
 	MicroTimer m_periodTimer;
-	int m_cpuLoad;
+	float m_cpuLoad;
 	QFile m_outputFile;
 
+	static const int s_detailCount = 4;                     // set to the actual number of used probes in Mixer.cpp
```

Comment:
If I understand correctly, this number, including the meaning of each index is tightly coupled with the implementation on the producer Mixer and consumer CPULoadWidget. 

In order to avoid confusions when only changing one part, I would suggest to be a bit more explicit. For example: 
* use a struct with getters:
```
struct CpuStatistics {
 uint32_t getTotalCpu();
 struct Stat {
     MicroTimer timer;
     detailLoad load; 
  } ;
  Stat mixer
  Stat xyz;
};

```

* have explicit constants for each index:


```
enum class CpuLoadTimer {
    Mixer,
    NumberOfTimers
};

std::array<Stat, NumberOfTimers> stats;

// Producer 
stats[CpuLoadTimer::Mixer] = xyz;

// Consumer
AddStat("Mixer", stats[CpuLoadTimer::Mixer] );
```

Reply:
I originally used a slightly more complex system, where you would first initialize each probe in the Mixer constructor with a string that describes what it measures, but eventually I removed that since I was worried about accidentally introducing dynamic allocations to the core when working with `QString` (which is what inevitably comes out of the `tr()` translation thing if I'm not mistaken). So I moved the strings to the GUI portions, since it wouldn't be changed often (if ever).

Using an enum instead of a magic constant is a good idea, thanks. I also thought about separate variables or a structure before, but I wanted to keep it in array so that I can process all detail stats in a single loop, avoiding lots of redundancy.